vsts-gn.yml

```diff
@@ -0,0 +1,72 @@
+resources:
+- repo: self
+phases:
+- phase: Build_Electron_via_GN
+  queue:
+    timeoutInMinutes: 180
+  steps:
+  - bash: |
+      git clone https://chromium.googlesource.com/chromium/tools/depot_tools.git
+      echo "##vso[task.setvariable variable=PATH]$PATH:`pwd`/depot_tools"
+    name: Setup_depot_tools
+
+  - bash: |
+      gclient config \
+        --name "src/electron" \
+        --unmanaged \
+        --cache-dir "/Users/electron/libcc_cache" \
+        "https://github.com/electron/electron"
+      gclient sync --with_branch_heads --with_tags
+    name: gclient_sync
+
+  - bash: |
+      cd src
+      export CHROMIUM_BUILDTOOLS_PATH=`pwd`/buildtools
+      export SCCACHE_WRAPPER="`pwd`/libchromiumcontent/tools/sccache/aad2120/mac/sccache"
+      $SCCACHE_WRAPPER --start-server --azure_container "$(SCCACHE_AZURE_BLOB_CONTAINER)" --azure_connection "$(SCCACHE_AZURE_CONNECTION_STRING)"
+      $SCCACHE_WRAPPER -s
+      echo "##vso[task.setvariable variable=SCCACHE_WRAPPER]$SCCACHE_WRAPPER"
+      echo "##vso[task.setvariable variable=CHROMIUM_BUILDTOOLS_PATH]`pwd`/buildtools"
+      gn gen out/Default --args='import("//electron/build/args/debug.gn") cc_wrapper="'"$SCCACHE_WRAPPER"'"'
+    name: GN_gen
+
+  - bash: |
+      cd src
+      ninja -C out/Default electron:electron_app
+    name: Ninja_build
+
+  - bash: |
+      $SCCACHE_WRAPPER -s
+      $SCCACHE_WRAPPER --stop-server
+    name: Check_sccache_stats
+
+  - bash: |
+      set +e
+      cd src
+      ninja -C out/Default electron/build/node:headers
+      export npm_config_nodedir="$PWD/out/Default/gen/node_headers"
+      (cd electron/spec && npm install)
+      ./out/Default/Electron.app/Contents/MacOS/Electron electron/spec --ci --enable-logging
+    name: Test
+
```

Comment:
* Does this need a `verify-ffmpeg` pass as in `vsts.yml`?

* Should we include `echo` tracers to the the `- bash: |` calls for consistency with `vsts.yml`?

Reply:
I haven't looked at `verify-ffmpeg` stuff anywhere yet. I assume we'll need that at some point but I don't think it blocks this first version.

I'm not sure what those echo tracers are for in `vsts.yml`, the UI seems to separate stuff just fine.